### PR TITLE
Update k8s tests and libraries to v1.26.0-rc.0

### PR DESCRIPTION
**Pull request by thorn3r**
Update k8s tests and libraries to v1.26.0-rc.0

Upstream changes included changing Ingress.LoadBalancerStatus from corev1.LoadBalancerStatus to networkingv1.IngressLoadBalancerStatus. This required the addition of 2 new factory funcs to convert slim.LoadBalancerIngress to networkingv1.IngressLoadBalancerIngress and another to convert LoadBalancerStatus to IngressLoadBalancerStatus in the slim client. See: https://github.com/kubernetes/kubernetes/pull/106242

Signed-off-by: Tim Horner <timothy.horner@isovalent.com>

This is a draft PR since I had some difficulties running the testsuites on my machine. Looking to leverage CI for that.

```release-note
Add support for k8s 1.26
```

### CODE REVIEW

File: Documentation/contributing/development/dev_setup.rst [hunk 1]
```diff
@@ -517,10 +517,10 @@ Minor version
    date with these changes.
 
 #  Update documentation files:
-   - Documentation/concepts/kubernetes/compatibility.rst
-   - Documentation/concepts/kubernetes/requirements.rst
    - Documentation/contributing/testing/e2e.rst
-   - Documentation/gettingstarted/istio.rst
+   - Documentation/network/istio.rst
+   - Documentation/network/kubernetes/compatibility.rst
+   - Documentation/network/kubernetes/requirements.rst
```
Comment (joestringer) [inline Documentation/contributing/development/dev_setup.rst:523]: nit: We would typically commit separate changes like this to another commit since it's unrelated to the overall k8s X.Y bump.

**Review by joestringer — APPROVED** [review]
LGTM for my codeowners, just one style nit.

It looks like the PR description instructions are a bit unclear, the line with `<!-- Enter ...` in the PR description needs to be removed. IIRC the tooling that generates the release notes doesn't know how to strip that line out. The code that does that is here if you think we should just make the tooling smarter: https://github.com/cilium/release/blob/3f899c03881388f4d38f0f2a098ab3674fda2a82/pkg/github/labels.go#L87.

**Review by aanm — APPROVED** [review]
Nice! :100:

File: .gitattributes [hunk 2]
```diff
@@ -9,3 +9,4 @@ Documentation/helm-values.rst linguist-generated
 Documentation/codeowners.rst linguist-generated
 Documentation/_static/* -diff
 *svg -diff
+pkg/k8s/client
```
Comment (aanm) [inline .gitattributes]: not all the directory should be ignored IIRC only these:
```
pkg/k8s/client/clientset/
pkg/k8s/client/informers/
pkg/k8s/client/listers/
```

File: Documentation/contributing/development/dev_setup.rst (hunk 1 above)
Comment (thorn3r) [reply Documentation/contributing/development/dev_setup.rst:523]: that makes sense, i'll keep that in mind for next time :+1:

File: .gitattributes (hunk 2 above)
Comment (thorn3r) [reply .gitattributes]: youre right good catch, i'll fix that up

**Comment by nbusseneau** [comment]
/test-1.26-net-next

**Review by rolinh — APPROVED** [review]

**Review by tklauser — APPROVED** [review]

**Comment by aanm** [comment]
/test-1.26-net-next

File: Documentation/network/kubernetes/compatibility.rst [hunk 3]
```diff
@@ -17,12 +17,12 @@ with Cilium. Older Kubernetes versions not listed in this table do not have
...
-+------------------------------------------------------------+---------------------------+----------------------------------+
-|                                                            |                           | ``cilium.io/v2`` has a           |
-| 1.16, 1.17, 1.18, 1.19, 1.20, 1.21, 1.22, 1.23, 1.24, 1.25 | * `networking.k8s.io/v1`_ | :term:`CustomResourceDefinition` |
-+------------------------------------------------------------+---------------------------+----------------------------------+
++------------------------------------------------------------------+---------------------------+----------------------------------+
+| k8s Version                                                      | k8s NetworkPolicy API     | CiliumNetworkPolicy              |
++------------------------------------------------------------------+---------------------------+----------------------------------+
+|                                                                  |                           | ``cilium.io/v2`` has a           |
+| 1.16, 1.17, 1.18, 1.19, 1.20, 1.21, 1.22, 1.23, 1.24, 1.25, 1.26 | * `networking.k8s.io/v1`_ | :term:`CustomResourceDefinition` |
```
Comment (nbusseneau) [inline Documentation/network/kubernetes/compatibility.rst:24]: When are we dropping support for old versions, please please pretty please? 🥺 @aanm @joestringer

**Review by nbusseneau — APPROVED** [review]

File: Documentation/network/kubernetes/compatibility.rst (hunk 3 above)
Comment (aanm) [reply Documentation/network/kubernetes/compatibility.rst:24]: I'm on the process of writing what concrete steps we will do before 1.13.

Comment (aanm) [reply Documentation/network/kubernetes/compatibility.rst:24]: @nbusseneau I'm on the process of writing what concrete steps we will do before 1.13.

**Review by nebril — APPROVED** [review]

**Comment by aanm** [comment]
/test-1.26-net-next
